mq.go

```diff
@@ -169,11 +174,18 @@ func (mq *mq) connect() error {
 }
 
 func (mq *mq) createConnection() (conn, error) {
+	mq.cluster.Do(func() { atomic.StoreInt32(&mq.cluster.currentNode, -1) })
```

Comment:
I think we can safely remove this optimization. It will simplify the code.
In case we care about the users without cluster setup and want to make more readable the code we can add DSNProvisioner interface with two different implementations for single node and cluster node setups.

Reply:
Not sure i understood you correctly. 
Do you mean completely remove everything related to cluster? and replace it with some sort of provisioner. 
Should we realize if it is cluster string or  single node connection and create here object for single/multiple node dsn provisioner?
I mean, this will hide logic and bring more complexity.

Reply:
Ok.